staging/src/k8s.io/kubectl/pkg/cmd/debug/profiles.go

```diff
@@ -64,54 +64,121 @@ type ProfileApplier interface {
 }
 
 // NewProfileApplier returns a new Options for the given profile name.
-func NewProfileApplier(profile string) (ProfileApplier, error) {
+func NewProfileApplier(profile string, keeps *KeepFields) (ProfileApplier, error) {
 	switch profile {
 	case ProfileLegacy:
-		return &legacyProfile{}, nil
+		return &legacyProfile{keeps}, nil
 	case ProfileGeneral:
-		return &generalProfile{}, nil
+		return &generalProfile{keeps}, nil
 	case ProfileBaseline:
-		return &baselineProfile{}, nil
+		return &baselineProfile{keeps}, nil
 	case ProfileRestricted:
-		return &restrictedProfile{}, nil
+		return &restrictedProfile{keeps}, nil
 	case ProfileNetadmin:
-		return &netadminProfile{}, nil
+		return &netadminProfile{keeps}, nil
 	case ProfileSysadmin:
-		return &sysadminProfile{}, nil
+		return &sysadminProfile{keeps}, nil
 	}
-
 	return nil, fmt.Errorf("unknown profile: %s", profile)
 }
 
 type legacyProfile struct {
+	*KeepFields
 }
 
 type generalProfile struct {
+	*KeepFields
 }
 
 type baselineProfile struct {
+	*KeepFields
 }
 
 type restrictedProfile struct {
+	*KeepFields
 }
 
 type netadminProfile struct {
+	*KeepFields
 }
 
 type sysadminProfile struct {
+	*KeepFields
 }
 
-func (p *legacyProfile) Apply(pod *corev1.Pod, containerName string, target runtime.Object) error {
-	switch target.(type) {
-	case *corev1.Pod:
-		// do nothing to the copied pod
-		return nil
-	case *corev1.Node:
-		mountRootPartition(pod, containerName)
-		useHostNamespaces(pod)
-		return nil
-	default:
-		return fmt.Errorf("the %s profile doesn't support objects of type %T", ProfileLegacy, target)
+type KeepFields struct {
+	labels         bool
+	annotations    bool
+	liveness       bool
+	readiness      bool
+	startup        bool
+	initContainers bool
+}
+
+func NewKeepFields() *KeepFields {
+	return &KeepFields{
+		initContainers: true,
+	}
+}
+
+func (keeps *KeepFields) WithKeepLabels(flag bool) *KeepFields {
+	keeps.labels = flag
+	return keeps
+}
+func (keeps *KeepFields) WithKeepAnnotations(flag bool) *KeepFields {
+	keeps.annotations = flag
+	return keeps
+}
+func (keeps *KeepFields) WithKeepLiveness(flag bool) *KeepFields {
+	keeps.liveness = flag
+	return keeps
+}
+func (keeps *KeepFields) WithKeepReadiness(flag bool) *KeepFields {
+	keeps.readiness = flag
+	return keeps
+}
+func (keeps *KeepFields) WithKeepStartup(flag bool) *KeepFields {
+	keeps.startup = flag
+	return keeps
+}
+func (keeps *KeepFields) WithKeepInitContainers(flag bool) *KeepFields {
+	keeps.initContainers = flag
+	return keeps
+}
+
+// removeLabels removes labels from the pod.
+func (keeps *KeepFields) removeLabels(p *corev1.Pod) {
```

Comment:
`KeepFields` is an exported type but it's all actual functions are not exported seems to be counter-intuitive. I know they are not used by anywhere else but what do you think about exporting the functions as well?

Reply:
The fix would likely eliminate the need to export types and functions in the first place.